### PR TITLE
MVSNet improvements pt 2 -- encoder/decoder refinement network

### PR DESCRIPTION
**Pull request by cpheinrich**
This PR introduces the RefineUNet which is a different architecture for the refinement network. One can select between refinement networks using the refinement_network_type flag. I also did a bit of cleanup.

This PR also includes the changes from -- https://github.com/ubiquity6/MVSNet/pull/2 which should be reviewed and merged first, so I'm opening this up as a WIP until that happens

### CODE REVIEW

File: mvsnet/cnn_wrapper/mvsnetworks.py [hunk 1]
```diff
@@ -176,5 +174,75 @@ def setup(self):
...
+        # (CH) I am experimenting with adding the residual to the 
+        # original depth estimate, rather than adding to normalized depth estimate
+        # followed by the denormalizing scaling. I think this is a good idea because the previous
+        # way of doing things constrained the output of refine_conv3 to only take on very small values
+        # which would lead to small gradients
+        #(self.feed('refine_conv3', 'depth_image')
+        # .add(name='refined_depth_image'))
+
+
+class RefineUNet(Network):
```
Comment (cpheinrich) [inline mvsnet/cnn_wrapper/mvsnetworks.py:186]: This is the new refinement network

**Review by indianajohn — APPROVED** [review]
You can edit this PR to use pt 1 as a base branch and then Github will show the relevant diff only - between this branch and that one.

Once again, not super prepared to comment on the approach here, but the code looks fine. Another set of eyes might be good.

**Review by miaoshasha — APPROVED** [review]
lgtm

**Review by cpheinrich — COMMENTED** [review]
> You can edit this PR to use pt 1 as a base branch and then Github will show the relevant diff only - between this branch and that one.
> 
> Once again, not super prepared to comment on the approach here, but the code looks fine. Another set of eyes might be good.

okay cool, and thanks for the tip about the PR diff!

**Pull request merged**